splash/src/main/kotlin/edu/artic/splash/SplashActivity.kt

```diff
@@ -106,20 +113,15 @@ class SplashActivity : BaseViewModelActivity<SplashViewModel>(), TextureView.Sur
     override fun onStart() {
         super.onStart()
         viewModel.navigateTo
+                .observeOn(AndroidSchedulers.mainThread())
                 .subscribe { navigation ->
                     when (navigation) {
                         is Navigate.Forward -> {
                             when (navigation.endpoint) {
                                 is SplashViewModel.NavigationEndpoint.StartVideo -> {
-                                    mMediaPlayer?.start()
-                                    splashChrome.postDelayed({
-                                        splashChrome.animate().alpha(0f).start()
-                                        val endpoint = navigation.endpoint as SplashViewModel.NavigationEndpoint.StartVideo
-                                        val displayDialog = endpoint.displayLanguageSettings
-                                        if (displayDialog) {
-                                            pauseVideo(4)
-                                        }
-                                    }, 200)
+                                    val endpoint = navigation.endpoint as SplashViewModel.NavigationEndpoint.StartVideo
```

Comment:
If you turn `navigation.endpoint` into a val _above_ the `when`, then you won't need an explicit cast here.